### PR TITLE
package-repositories: draft spec and cleanup

### PR DESCRIPTION
**Pull request by cjp256**
specifications: add package-repositories draft spec

schema: remove exposure of reserved name in package-repositories

cli: gate package-repositories on --enable-experimental-package-repositories

### CODE REVIEW

File: specifications/package-repositories.org [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+GPG key ID: <key-id>
+GPG key server: <key-server>
+#+END_EXAMPLE
+
+* GPG Keyring handling
+
+All GPG keys will be imported into a new keyring:
+=/etc/apt/trusted.gpg.d/snapcraft.gpg=
+
+* Implementation
```
Comment (sergiusens) [inline specifications/package-repositories.org:310]: Why here and not in /todo.org?

**Comment by cjp256** [comment]
Oh I'm a goof.

File: snapcraft/cli/_options.py [hunk 2]
```diff
@@ -337,3 +347,7 @@ def apply_host_provider_flags(build_provider_flags: Dict[str, str]) -> None:
                 os.environ.pop(key)
         else:
             os.environ[key] = str(value)
+
+    # Log any experimental flags in use.
+    if build_provider_flags.get("SNAPCRAFT_ENABLE_EXPERIMENTAL_PACKAGE_REPOSITORIES"):
+        logger.warning("*EXPERIMENTAL* package-repositories in use")
```
Comment (sergiusens) [inline snapcraft/cli/_options.py:353]: please use echo.warn here.

File: snapcraft/internal/project_loader/_config.py [hunk 3]
```diff
@@ -219,6 +220,14 @@ def __init__(self, project: project.Project) -> None:
             properties=self.data, project=project
         )
 
+        if project._snap_meta.package_repositories and not os.getenv(
```
Comment (sergiusens) [inline snapcraft/internal/project_loader/_config.py:223]: is this a safeguard? sys.exit(1) is not friendly for testing or figuring things out. If this is just a guard a RuntimeError could be better

Comment (sergiusens) [reply snapcraft/internal/project_loader/_config.py:223]: If not, project sanity checks
and lastly, a new exception

File: snapcraft/cli/_options.py (hunk 2 above)
Comment (cjp256) [reply snapcraft/cli/_options.py:353]: will do

File: snapcraft/internal/project_loader/_config.py (hunk 3 above)
Comment (cjp256) [reply snapcraft/internal/project_loader/_config.py:223]: Unfortunately `_is_legacy` will catch the exception here.

Comment (cjp256) [reply snapcraft/internal/project_loader/_config.py]: Moved into conduct_project_sanity_check, that works.

File: snapcraft/cli/_options.py (hunk 2 above)
Comment (cjp256) [reply snapcraft/cli/_options.py]: done

**Pull request merged**